### DISCUSSION TITLE
Open Drawer from Upload Field

### DISCUSSION THREAD

**Discussion by mattddean** [payloadcms/payload#3042] [Feature Requests & Ideas]
It would be nice if there were a way to open a drawer pointing to an uploaded file.

We're using the S3 plugin, and after setting an image in an upload field, the link on the upload field takes us straight to the file on the remote server rather than to the uploaded file document in Payload.

This means that something like editing Alt Text for an image requires leaving the current collection and finding the uploaded file in the images collection.

**Comment by jacobsfletch** [comment]
Hey @mattddean! Totally see your point here. Coming up soon on our roadmap is a reusable component library, and this would certainly fall under that scope: https://github.com/payloadcms/payload/discussions/293. I threw a comment in there so we can ensure it happens: https://github.com/payloadcms/payload/discussions/293#discussioncomment-6505110

**Reply by mattddean** [reply]
I'm not sure I understand; we're not using a custom component, just an upload collection. Seems like this would be a good built-in feature, but maybe I'm misunderstanding. The filename here links to the external file, not to the internal document.

<img width="1256" alt="Screenshot 2023-07-20 at 9 01 17 PM" src="https://github.com/payloadcms/payload/assets/29106809/df1e1b9b-94a1-4230-9ed2-044a2693029e">

**Reply by jacobsfletch** [reply]
Ahh ok that's making more sense...it's late. Yea this is greatly needed.

**Reply by mattddean** [reply]
Yeah, it was late, I'm sorry for not describing it well. Thanks for listening!

**Comment by JessRynkar** [comment]
@mattddean @jacobsfletch I've added this to the roadmap (priority 1 ☝️ ) and will build this feature next week, stay tuned...

**Reply by mattddean** [reply]
Wow cool! There's no rush; it's just a nice-to-have.

**Comment by JarrodMFlesch** [comment]
Cleaning up. This was added a while back! https://github.com/payloadcms/payload/pull/3108
